lib/tasks/dev.rake

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+namespace :dev do
+  require 'factory_bot_rails'
+  require 'faker'
+  desc 'rake task'
+  task :generate_data, [:number] => :environment do |_, args|
+    @args = {}
+    @args[:number] = (args.number || 1).to_i
+    language_codes = %w[vn en fr jp]
+    programming_language_names = %w[ruby php java javascript pithon nodejs]
+    @args[:number].times do
+      email = loop do
+        email = Faker::Internet.email
+        break email unless Developer.exists?(email: email)
+      end
+      developer = FactoryBot.create(:developer, email: email)
+      languages = language_codes.sample(rand(4))
+      languages.each do |lang|
+        language = Language.find_or_create_by(code: lang)
```

Comment:
Considering the numbers of developers/languages  are 100/4 by default, the line (accessing the languages table) looks very inefficient.
Basically, creating records in the language table should be nothing to do with the developers table.